parsers/wiki_parser.py

```diff
@@ -0,0 +1,291 @@
+    populate_chat_wheel()
+
+
+def populate_hero_responses():
+    """Method that populates hero responses (as well as Arcana voice packs and Announcer packs) from Gamepedia.
+    First fetches all Pages in Responses category, then source for each page.
+    Populates Responses table and Hero table from processed response, original response, link and hero name.
+    """
+    pages = pages_for_category(RESPONSES_CATEGORY)
+    for page in pages:
+        if is_hero_type(page):
+            # page points to hero responses
+            hero_name = get_hero_name(page)
+        else:
+            # page points to voice pack, announcer or shopkeeper responses
+            hero_name = page
+
+        responses_source = requests.get(url=URL_DOMAIN + '/' + page, params={'action': 'raw'}).text
+
+        response_link_list = create_responses_text_and_link_list(responses_source=responses_source)
+        # Note: Save all responses to the db. Apply single word and common words filter on comments and submission text
+        # not while saving responses
+        db_api.add_hero_and_responses(hero_name=hero_name, response_link_list=response_link_list)
+
+
+def pages_for_category(category_name):
+    """Method that returns a list of pages for a given Wiki category.
+
+    :param category_name: returns all category members in json response from mediawiki API.
+    :return: list of all `pages` in the given category.
+    """
+    params = get_params_for_category_api(category_name)
+    json_response = requests.get(url=API_PATH, params=params).text
+
+    pages = []
+
+    parsed_json = json.loads(json_response)
+    for category_members in parsed_json['query']['categorymembers']:
+        title = category_members['title']
+        pages.append(title)
+
+    return pages
+
+
+def get_params_for_category_api(category):
+    """Method to get `GET` parameters for querying MediaWiki for category details.
+
+    :param category: category name to be passed in params.
+    :return: GET parameters `params`
+    """
+    params = CATEGORY_API_PARAMS.copy()
+    params['cmtitle'] = 'Category:' + category
+    return params
+
+
+def get_params_for_files_api(files):
+    """Method to get `GET` parameters for querying MediaWiki for details for multiple files.
+    Uses pipe character `|` to include multiple files. Currently MediaWiki limits number of files to 50.
+    If files list is empty, leave `File` parameter empty.
+
+    :param files: list of file names to be passed in params.
+    :return: GET parameters `params`.
+    """
+    params = FILE_API_PARAMS.copy()
+    if files:
+        titles = 'File:' + '|File:'.join(files)
+    else:
+        titles = ''
+    params['titles'] = titles
+    return params
+
+
+def is_hero_type(page):
+    """Method to check if page belongs to a hero, creep-hero(Warlock's Golem).
+
+    :param page: Page name as string.
+    :return: True if page belongs to hero else False
+    """
+    if '/Responses' in page:
```

Comment:
Isn' the whole if-else actually reduced to the code below?

    return '/Responses' in page

Reply:
You're right. Fixed in d266d930fbd4885d99aa0bd34300009ef2e3340c